shell/platform/android/io/flutter/embedding/engine/systemchannels/TextInputChannel.java

```diff
@@ -186,6 +188,18 @@ public void requestExistingInputState() {
     state.put("composingExtent", composingEnd);
     return state;
   }
+
+  private static HashMap<Object, Object> createEditingDeltaJSON(
+      ArrayList<TextEditingDelta> batchDeltas) {
+    HashMap<Object, Object> state = new HashMap<>();
+
+    JSONArray deltas = new JSONArray();
+    for (TextEditingDelta delta : batchDeltas) {
+      deltas.put(delta.toJSON());
+    }
+    state.put("batchDeltas", deltas);
```

Comment:
I made a comment in the framework about how we send these batched deltas: https://github.com/flutter/flutter/pull/88477#discussion_r704838312

Reply:
This is all good now, we always send an array of batchDeltas.